packages/languages/src/common/console-window.ts

```diff
@@ -5,25 +5,35 @@
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
 
-import { injectable } from 'inversify';
+import { injectable, inject } from 'inversify';
+import { MessageService } from '@theia/core/lib/common';
 import { MessageActionItem, MessageType } from 'vscode-base-languageclient/lib/protocol';
 import { Window, OutputChannel } from 'vscode-base-languageclient/lib/services';
 
 @injectable()
 export class ConsoleWindow implements Window {
     protected readonly channels = new Map<string, OutputChannel>();
+    constructor( @inject(MessageService) protected readonly messageService: MessageService) { }
     showMessage<T extends MessageActionItem>(type: MessageType, message: string, ...actions: T[]): Thenable<T | undefined> {
+        const originalActions = new Map((actions || []).map(action => [action.title, action] as [string, T]));
+        const actionTitles = (actions || []).map(action => action.title);
+        const mapActionType: (result: string | undefined) => (T | undefined) = result => {
+            if (!!result) {
+                return originalActions.get(result);
+            }
+            return undefined;
+        };
         if (type === MessageType.Error) {
-            console.error(message);
```

Comment:
I think we should rename `ConsoleWindow` to something else, like `WindowImpl`

Reply:
and the file as well

Reply:
done